app/src/main/java/com/kickstarter/viewmodels/SearchViewModel.java

```diff
@@ -135,12 +143,22 @@ public ViewModel(final @NonNull Environment environment) {
         });
 
       params
-        .compose(takePairWhen(pageCount))
-        .filter(paramsAndPageCount -> paramsAndPageCount.first.sort() != defaultSort && IntegerUtils.intValueOrZero(paramsAndPageCount.second) == 1)
-        .map(paramsAndPageCount -> paramsAndPageCount.first)
-        .observeOn(Schedulers.io())
-        .compose(bindToLifecycle())
-        .subscribe(this.lake::trackSearchResultsLoaded);
+          .compose(takePairWhen(pageCount))
+          .compose(takePairWhen(this.discoverEnvelope))
```

Comment:
Instead of using `.takePairWhen` twice and creating a pair within a pair, consider using `Observable.zip` for all three streams and utilizing kotlin's `Triple` object. For example:

```     
      Observable.zip(
              params,
              pageCount,
              this.discoverEnvelop,
              (params, pageCount, discoverEnvelop) -> new Triple<DiscoveryParams, Integer, DiscoverEnvelop>(params, pageCount, discoverEnvelop));
```

This is the verbose example, but the lambda can be replaced with a method reference as long as you explicitly define what the parameter types of the `Triple` are before you create the method reference, otherwise the compiler won't know what type of items are in the triple and you won't have access to any of their class methods. I think it looks a bit cleaner and more readable. Inevitably, it should look something like this:

```Observable.zip(params, pageCount, this.discoverEnvelop, Triple::new)```

Reply:
Closing this because we plan on creating a `combineLatestTriple()` transformer in the near future that would solve this.